src/main/java/net/sf/jabref/model/database/BibDatabase.java

```diff
@@ -120,6 +120,20 @@ public boolean containsEntryWithId(String id) {
         return internalIDs.contains(id);
     }
 
+    /**
+     * Returns whether an entry with the given bibtex key exists.
+     */
+    public synchronized boolean containsEntryWithKey(String key) {
+        int keyHash = key.hashCode(); // key hash for better performance
```

Comment:
MIssed this comment, but getEntryByKeyOptional was introduced in #1568. It is an intermediate solution before figuring out the final conversion problem.

Makes sense to return the first, so I'll implement that later on tonight.